utils/forest_utils/src/io/progress_bar.rs

```diff
@@ -42,47 +42,63 @@ static PROGRESS_BAR_VISIBILITY: RwLock<ProgressBarVisibility> =
 
 /// Progress bar wrapper, allows suppressing progress bars.
 pub struct ProgressBar {
-    inner: RefCell<pbr::ProgressBar<Stdout>>,
+    inner: RefCell<IndicatifProgressBar>,
     display: bool,
+    size: u64,
 }
 
 impl ProgressBar {
     pub fn new(size: u64) -> Self {
         Self {
-            inner: RefCell::new(pbr::ProgressBar::new(size)),
+            inner: RefCell::new(IndicatifProgressBar::with_draw_target(
+                Some(size),
+                ProgressDrawTarget::stdout(),
+            )),
             display: Self::should_display(),
+            size,
         }
     }
 
     pub fn message(&self, message: &str) {
         if self.display {
-            self.inner.borrow_mut().message(message);
+            self.inner.borrow_mut().set_message(message.to_string());
         }
     }
 
-    pub fn set_max_refresh_rate(&self, w: Option<Duration>) {
+    pub fn set_max_refresh_rate(&mut self, w: u8) {
         if self.display {
-            self.inner.borrow_mut().set_max_refresh_rate(w);
+            self.inner = RefCell::new(IndicatifProgressBar::with_draw_target(
+                Some(self.size),
+                ProgressDrawTarget::term(Term::buffered_stdout(), w),
+            ));
         }
     }
 
     pub fn add(&self, i: u64) -> u64 {
         if self.display {
-            self.inner.borrow_mut().add(i)
+            self.inner.borrow_mut().inc(i);
+            self.inner.borrow_mut().position()
         } else {
             0
         }
     }
 
-    pub fn set_units(&self, u: Units) {
+    pub fn set_bytes(&self) {
         if self.display {
-            self.inner.borrow_mut().set_units(u)
+            self.inner.borrow_mut().set_style(
+                ProgressStyle::with_template(
+                    "{spinner:.green} [{wide_bar:.cyan/blue}] {bytes}/{total_bytes} ({eta})",
```

Comment:
The previous progress bar looked like this:
`Downloading snapshot 107.42 MB / 423.67 MB [===>--------------] 25.35 % 35.79 MB/s 9s`

And the new progress bar looks like this:
`⠤ [##>------------] 75.60 MiB/423.67 MiB (9s)`

Could you change the format to include the message, the progress as a percentage, and the rate of progress?

Reply:
Fixed accordingly